modules/market/src/main/java/com/opengamma/strata/market/curve/CurveGroupDefinitionBuilder.java

```diff
@@ -65,7 +68,32 @@ public CurveGroupDefinitionBuilder addDiscountCurve(
     ArgChecker.notNull(curveDefinition, "curveDefinition");
     ArgChecker.notNull(currency, "currency");
     CurveGroupEntry entry = CurveGroupEntry.builder()
-        .curveDefinition(curveDefinition)
+        .curveName(curveDefinition.getName())
+        .discountCurrencies(ImmutableSet.copyOf(Lists.asList(currency, otherCurrencies)))
+        .build();
+    return merge(entry, curveDefinition);
+  }
+
+  /**
+   * Adds the definition of a discount curve to the curve group definition.
+   * <p>
+   * A curve added with this method cannot be calibrated by the market data system as it does not include
+   * a curve definition. It is intended to be used with curves which are supplied by the user.
+   *
+   * @param curveName  the name of the curve
+   * @param otherCurrencies  additional currencies for which the curve can provide discount factors
+   * @param currency  the currency for which the curve provides discount rates
+   * @return this builder
+   */
+  public CurveGroupDefinitionBuilder addDiscountCurve(
```

Comment:
I can see your point, but I think both of those alternative names actually make it more confusing. I think `addXxxCurve` communicates what's happening pretty well. You're telling the curve group that it contains a curve with a specified name. That feels like an `add` operation to me.

If the other methods were actually adding `Curve` objects I'd see the need to change it, but they are only adding a curve definition so I don't think the distinction is that great.

I think the the current names aren't perfect but I think they're the lesser of two evils.